Sources/TuistGraph/Graph/GraphDependency.swift

```diff
@@ -23,7 +23,8 @@ public enum GraphDependency: Hashable, CustomStringConvertible, Comparable, Coda
         primaryBinaryPath: AbsolutePath,
         linking: BinaryLinking,
         mergeable: Bool,
-        status: FrameworkStatus
+        status: FrameworkStatus,
+        macroPath: AbsolutePath?
```

Comment:
Not necessarily specific to this PR but I wonder if we should create a struct for the `XCFrameworkMetadata` (or reuse the one we already have in the `XCFrameworkMetadataProvider`. I feel having so many associated values for a case makes it harder to work with:
- Often, you only want a one or two associated values, leading to cases like `case let .xcframework(path, _, _, _, _, _, _):` with a lot of underscores
- Adding a new case means you need to update a bunch of files, even though you might need the value only a single place

Might be worth a subsequent refactor.